lib/cronitor.rb

```diff
@@ -1,138 +1,105 @@
+  def self.TYPE_JOB
+    'job'
   end
 
-  def create
-    uri = URI.parse "#{API_URL}/monitors"
-
-    http = Net::HTTP.new uri.host, uri.port
-    http.use_ssl = uri.scheme == 'https'
-
-    request = Net::HTTP::Post.new uri.path, default_headers
-    request.basic_auth token, nil
-    request.content_type = 'application/json'
-    request.body = JSON.generate opts
-
-    response = http.request request
-
-    @code = JSON.parse(response.body).fetch 'code' if valid? response
+  def self.TYPE_EVENT
+    'event'
   end
 
-  def exists?(name)
-    uri = URI.parse "#{API_URL}/monitors/#{URI.escape name}"
-
-    http = Net::HTTP.new uri.host, uri.port
-    http.use_ssl = uri.scheme == 'https'
-
-    request = Net::HTTP::Get.new uri.path, default_headers
-    request.basic_auth token, nil
-
-    response = http.request request
-
-    return false unless response.is_a? Net::HTTPSuccess
-
-    @code = JSON.parse(response.body).fetch 'code'
-    @opts = JSON.parse(response.body)
+  def self.TYPE_SYNTHETIC
+    'synthetic'
+  end
 
-    true
+  def self.MONITOR_TYPES
+    [self.TYPE_JOB, self.TYPE_EVENT, self.TYPE_SYNTHETIC]
   end
 
-  def ping(type, msg = nil)
-    uri = URI.parse "#{PING_URL}/#{URI.escape code}/#{URI.escape type}"
-    if %w[run complete fail].include?(type) && !msg.nil?
-      uri.query = URI.encode_www_form 'msg' => msg
+  def self.read_config(path=nil, output: false)
+    Cronitor.config = path || Cronitor.config
+    unless Cronitor.config
+        raise ConfigurationError.new(
+          "Must include a path by setting Cronitor.config or passing a path to read_config e.g. Cronitor.read_config('./cronitor.yaml')"
+        )
     end
 
-    http = Net::HTTP.new uri.host, uri.port
-    http.use_ssl = uri.scheme == 'https'
+    conf = YAML.load(File.read(Cronitor.config))
 
-    request = Net::HTTP::Get.new uri, default_headers
+    conf.each{ |k, v|
+      raise ConfigurationError.new("Invalid configuration variable: #{k}") unless self.YAML_KEYS.include?(k)
+    }
 
-    response = http.request request
-
-    valid? response
-  end
-
-  private
+    if conf[:api_key]
+      Cronitor.api_key = conf[:api_key]
+    end
+    if conf[:api_version]
+      Cronitor.api_version = conf[:api_version]
+    end
+    if conf[:environment]
+      Cronitor.environment = conf[:environment]
+    end
 
-  def valid?(response)
-    return true if response.is_a? Net::HTTPSuccess
+    return unless output
 
-    msg = if response.content_type.match? 'json'
-            error_msg JSON.parse(response.body)
-          else
-            "Something else has gone awry. HTTP status: #{response.code}"
-          end
+    monitors = []
+    self.MONITOR_TYPES.each do |t|
+      plural_t = "#{t}s"
+      to_parse = conf[t] || conf[plural_t] || nil
+      return unless to_parse
 
-    raise Cronitor::Error, msg
-  end
+      if !to_parse.is_a?(Hash)
+        raise ConfigurationError.new("A dict of with keys corresponding to monitor keys is expected.")
```

Comment:
I’m sensing some more Python-ness 🙂 We should stick with the Ruby lingo, and say “A hash of…” here.